src/crypto/secretbox/xsalsa20poly1305.rs

```diff
@@ -71,6 +71,25 @@ pub fn seal(m: &[u8],
     c
 }
 
+/// `seal_detached_inplace()` encrypts and authenticates a message `m` using a secret key `k` and a
+/// nonce `n`.  `m` is encrypted in place, so after this function returns it will contain the
+/// ciphertext.  The detached authentication tag is written to `mac`.
+pub fn seal_detached_inplace(m: &mut[u8],
```

Comment:
I thought about using `Tag` at first, but the problem I ran into is that I don't know a safe way to get an `&Tag` from an `&[u8; MACBYTES]`. That's not a big deal if the tag is just sitting on the stack by itself, but if the caller wants it to be the first 16-bytes of a combined-mode buffer, I don't know if we'll be able to do that. Instead the caller/wrapper would have to do a second copy from their `Tag` on the stack to their buffer, after calling this function. (Or the other way around, for decryption.)

We could do a `transmute`, but that seemed like serious unsafety overkill, and I'm not even sure it's correct if the compiler chooses to do interesting things with padding in the newtype.

Any other ideas? Maybe having the caller do a copy isn't so bad, since we know it's only 64 bytes?